model_compression_toolkit/core/common/substitutions/shift_negative_activation.py

```diff
@@ -476,16 +421,15 @@ def shift_negative_function(graph: Graph,
 
     if len(previous_nodes) == 1:
         fused_candidates = [previous_nodes[0], non_linear_node, add_node]
+        graph.fusing_info.fusing_patterns.append([n.type for n in fused_candidates])
```

Comment:
is this needed EVERY SNC call? isn't it the same pattern most of the time?
maybe `fusing_patterns` should be a set

Reply:
The graph can contain different patterns (if, for example, it has conv->swish and dw->prelu, different patterns are needed).

Reply:
of course, but in most cases,1 or 2 patterns will exist in a model